src/main/java/net/countercraft/movecraft/combat/features/tracking/BlockBehaviorOverride.java

```diff
@@ -0,0 +1,420 @@
+package net.countercraft.movecraft.combat.features.tracking;
+
+import it.unimi.dsi.fastutil.Pair;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import net.countercraft.movecraft.combat.MovecraftCombat;
+import net.countercraft.movecraft.util.Tags;
+import org.apache.commons.lang.reflect.FieldUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class BlockBehaviorOverride {
+
+    public record BlockOverride(
+            Optional<Float> blastResistanceOverride,
+            Optional<Integer> burnOddity,
+            Optional<Integer> igniteOddity,
+            Optional<Float> vanillaBlastResistanceOverride,
+            Optional<Integer> vanillaBurnOddity,
+            Optional<Integer> vanillaIgniteOddity
```

Comment:
I've not actually tested this assumption, so it'd be good to know.

Reply:
The burn oddities? Not really, those are stored within two maps in the fireblock class and we replace those values, so unless the server restarts, we can't access them. The blast resistance? Ye, that one is stored on the material.

Reply:
Problem is, what vanilla does is it runs FireBlock#bootstrap() at some point when launching. That basically calls a "setFlammable(burnOddity, igniteOddity, Block)" for quite a few blocks. That function writes the given values to two different maps, which i access here. Problem is, that the original values are only given in the arguments when that function is called, so that's why we need to cache them. Alternatively, we could access both maps and clear then and then call the bootstrap method again.

Reply:
I'm ok with storing the two values then, I just didn't want to if it was easy to access.